modules/platforms/cpp/ignite/odbc/query/column_metadata_query.h

```diff
@@ -0,0 +1,188 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "ignite/odbc/query/query.h"
+#include "ignite/odbc/meta/column_meta.h"
+
+namespace ignite
+{
+
+/** Connection forward-declaration. */
+class sql_connection;
+
+/**
+ * Column metadata.
+ */
+struct odbc_column_meta {
+    /**
+     * Default constructor.
+     */
+    odbc_column_meta() = default;
+
+    /** Label. */
+    std::string label;
+
+    /** Schema name. */
+    std::optional<std::string> schema;
+
+    /** Table name. */
+    std::optional<std::string> table;
+
+    /** Column name. */
+    std::optional<std::string> column;
+
+    /** Data type. */
+    ignite_type data_type{ignite_type::BOOLEAN};
```

Comment:
Why `ignite_type::BOOLEAN`?

Reply:
Good catch. I was going to put `UNDEFINED`. Probably just a slip.